modules/bastion/main.tf

```diff
@@ -0,0 +1,319 @@
+/*
+Copyright 2025 Chainguard, Inc.
+SPDX-License-Identifier: Apache-2.0
+*/
+
+locals {
+  // Derive region from zone (e.g. us-central1-a -> us-central1)
+  region        = join("-", slice(split("-", var.zone), 0, 2))
+  instance_name = substr(var.name, 0, 63) # GCE name limit
+  instance_tag  = var.name
+
+  // Labels
+  default_labels = {
+    bastion = var.name
+  }
+
+  squad_label = {
+    squad = var.squad
+    team  = var.squad
+  }
+
+  merged_labels = merge(local.default_labels, local.squad_label)
+
+  // Split patch time HH:MM into components
+  patch_hour   = tonumber(split(":", var.patch_time_utc)[0])
+  patch_minute = tonumber(split(":", var.patch_time_utc)[1])
+
+  bastion_startup = join("\n", compact([
+    <<-SCRIPT
+      #!/usr/bin/env bash
+      set -euo pipefail
+    SCRIPT
+    ,
+
+    <<-COMMON
+      # ------------------------------------------------------
+      # Install Google Cloud Ops Agent for logging & metrics
+      # ------------------------------------------------------
+      curl -sSO https://dl.google.com/cloudagents/google-cloud-ops-agent-installer
+      bash google-cloud-ops-agent-installer --also-install
+
+      # ------------------------------------------------------
+      # Enable auditd to record all commands & auth events
+      # ------------------------------------------------------
+      apt-get update -y
+      apt-get install -y auditd
+      systemctl enable --now auditd.service
+
+      cat > /etc/audit/rules.d/99-bastion.rules <<'AUDIT'
+      -w /usr/bin/ -p x -k execs
+      -w /bin/ -p x -k execs
+      -a always,exit -F arch=b64 -S execve -k execs
+      AUDIT
+
+      augenrules --load
+    COMMON
+    ,
+
+    var.postgres_conn_name != "" ? <<-POSTGRES
+      # ------------------------------------------------------
+      # Install Cloud SQL Auth Proxy v2 (PostgreSQL)
+      # ------------------------------------------------------
+      curl -sSL -o /usr/local/bin/cloud-sql-proxy \
+        https://storage.googleapis.com/cloud-sql-connectors/cloud-sql-proxy/v2.8.2/cloud-sql-proxy.linux.amd64
```

Comment:
is this the latest?

how we will track the releases and update when needed?

Reply:
but we can do in a follow up not a problem for this PR

Reply:
it's v2.16 in the latest commit, this one is outdated